invenio/modules/deposit/cache.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015 CERN.
+#
+# Invenio is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""Implementation of allowed deposit type caching."""
+
+from invenio.legacy.miscutil.data_cacher import DataCacher, DataCacherProxy
+
+
+class DepositTypeDataCacher(DataCacher):
+    def __init__(self):
+        def cache_filler():
+            from invenio.modules.access.control import acc_get_action_id
+            from invenio.modules.access.models import AccAuthorization, \
+                AccARGUMENT
+
+            action_id = acc_get_action_id('usedeposit')
+
+            return [auth[0] for auth in AccAuthorization.query.join(
```

Comment:
@jirikuncar yes, I think this fix or basically returning a list instead of a set in `def cache_filler()` should solve it. Everything seems to work, apart from the fact that all users get authorized, because of this function:

https://github.com/inveniosoftware/invenio/blob/master/invenio/modules/access/firerole.py#L194-L273

It authorizes everyone by default, but I understand it's configurable and it's not a problem with my code, am I right?